src/check_ast.rs

```diff
@@ -1240,6 +1240,23 @@ where
                 args,
                 keywords,
             } => {
+                // pyflakes
+                if let ExprKind::Attribute { value, attr, .. } = &func.node {
+                    if let ExprKind::Constant {
+                        value: Constant::Str(value),
+                        ..
+                    } = &value.node
+                    {
+                        if attr == "format" {
+                            // "...".format(...) call
+                            if self.settings.enabled.contains(&CheckCode::F521) {
```

Comment:
We can probably move this to the top of the block, like line 1244?

Reply:
This is mostly pre-emptive laziness on my part - once we add the 5 checks I was going to write it like:
```rust
if string_dot_format_call {
    if F521_enabled {...}
    if F522_enabled {...}
    if F523_enabled {...}
    if F524_enabled {...}
    if F525_enabled {...}
}
```

rather than
```rust
if F521_enabled {
    if string_dot_format_call { ... }
}
if F522_enabled {
    if string_dot_format_call { ... }
}
...
```

I'm OK doing it the second way if you think that's better (would just place the repeated logic in a helper function)?

Reply:
Oh, yes, the pattern you were intending makes sense to me.